### PR TITLE
lex: add examples and add {Free,Net,Open}BSD pages

### PR DESCRIPTION
**Pull request by vitorhcl**
- [X] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [X] The page(s) have at most 8 examples.
- [X] The page description(s) have links to documentation or a homepage.
- [X] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [X] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

### CODE REVIEW

**Comment by vitorhcl** [comment]
I have an idea, as in this case the options in all platforms are the same we could leave flex (as a more complete page) in common and leave lex as the POSIX page, referencing flex in this page, what do you think?

**Comment by kbdharun** [comment]
> I have an idea, as in this case the options in all platforms are the same we could leave flex (as a more complete page) in common and leave lex as the POSIX page, referencing flex in this page, what do you think?

Sure, that would work too.

File: pages/common/lex.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 > Given the specification for a lexical analyzer, generates C code implementing it.
 > More information: <https://manned.org/lex.1>.
 
-- Generate an analyzer from a Lex file:
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
```
Comment (gutjuri) [inline pages/common/lex.md:7]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:
```

File: pages/freebsd/lex.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# lex
+
+> Lexical analyzer generator.
+> Given the specification for a lexical analyzer, generates C code implementing it.
+> More information: <https://man.freebsd.org/cgi/man.cgi?query=lex&sektion=1>.
+
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
```
Comment (gutjuri) [inline pages/freebsd/lex.md:7]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:
```

This suggestion should also be applied to the other platform files.

File: pages/freebsd/lex.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+`lex {{analyzer.l}}`
+
+- Write analyzer to `s`[`t`]`dout`:
```
Comment (gutjuri) [inline pages/freebsd/lex.md:11]:
```suggestion
- Write the analyzer to `s`[`t`]`dout`:
```

**Review by gutjuri — CHANGES_REQUESTED** [review]
Thanks, I have some suggestions

File: pages/freebsd/lex.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+`lex -{-stdout|t} {{analyzer.l}}`
```
Comment (kbdharun) [inline pages/freebsd/lex.md:13]:
```suggestion
`lex -{{-stdout|t}} {{analyzer.l}}`
```

File: pages/linux/lex.md [hunk 5]
```diff
@@ -4,14 +4,22 @@
...
 > More information: <https://manned.org/lex.1>.
 
-- Generate an analyzer from a Lex file:
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
 
 `lex {{analyzer.l}}`
 
+- Write analyzer to `s`[`t`]`dout`:
+
+`lex -{-stdout|t} {{analyzer.l}}`
```
Comment (kbdharun) [inline pages/linux/lex.md:13]:
```suggestion
`lex -{{-stdout|t}} {{analyzer.l}}`
```

File: pages/netbsd/lex.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+> Lexical analyzer generator.
+> Given the specification for a lexical analyzer, generates C code implementing it.
+> More information: <https://man.netbsd.org/NetBSD-9.3/lex.1>.
+
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
+
+`lex {{analyzer.l}}`
+
+- Write analyzer to `s`[`t`]`dout`:
```
Comment (kbdharun) [inline pages/netbsd/lex.md:11]:
```suggestion
- Write analyzer to `stdout`:
```

File: pages/netbsd/lex.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+`lex -{-stdout|t} {{analyzer.l}}`
```
Comment (kbdharun) [inline pages/netbsd/lex.md:13]:
```suggestion
`lex -{{-stdout|t}} {{analyzer.l}}`
```

File: pages/linux/lex.md [hunk 8]
```diff
@@ -4,14 +4,22 @@
 > Given the specification for a lexical analyzer, generates C code implementing it.
 > More information: <https://manned.org/lex.1>.
 
-- Generate an analyzer from a Lex file:
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
```
Comment (kbdharun) [inline pages/linux/lex.md:7]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:
```

File: pages/netbsd/lex.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# lex
+
+> Lexical analyzer generator.
+> Given the specification for a lexical analyzer, generates C code implementing it.
+> More information: <https://man.netbsd.org/NetBSD-9.3/lex.1>.
+
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
```
Comment (kbdharun) [inline pages/netbsd/lex.md:7]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:
```

File: pages/openbsd/lex.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+> Lexical analyzer generator.
+> Given the specification for a lexical analyzer, generates C code implementing it.
+> More information: <https://man.openbsd.org/flex>.
+
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
+
+`lex {{analyzer.l}}`
+
+- Write analyzer to `s`[`t`]`dout`:
```
Comment (kbdharun) [inline pages/openbsd/lex.md:11]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:

`lex {{analyzer.l}}`

- Write analyzer to `stdout`:
```

File: pages/osx/lex.md [hunk 11]
```diff
@@ -4,14 +4,22 @@
 > Given the specification for a lexical analyzer, generates C code implementing it.
 > More information: <https://keith.github.io/xcode-man-pages/lex.1.html>.
 
-- Generate an analyzer from a Lex file:
+- Generate an analyzer from a Lex file to the file `lex.yy.c`:
 
 `lex {{analyzer.l}}`
 
+- Write analyzer to `s`[`t`]`dout`:
```
Comment (kbdharun) [inline pages/osx/lex.md:11]:
```suggestion
- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:

`lex {{analyzer.l}}`

- Write analyzer to `stdout`:
```

File: pages/osx/lex.md [hunk 12]
```diff
@@ -4,14 +4,22 @@
...
+
+`lex -{-stdout|t} {{analyzer.l}}`
```
Comment (kbdharun) [inline pages/osx/lex.md:13]:
```suggestion
`lex -{{-stdout|t}} {{analyzer.l}}`
```

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page. (continuing @gutjuri's review). Additionally, we can remove the mnemonic highlighting from the `stdout` string.

File: pages/linux/lex.md [hunk 13]
```diff
@@ -4,14 +4,22 @@
 > Given the specification for a lexical analyzer, generates C code implementing it.
 > More information: <https://manned.org/lex.1>.
 
-- Generate an analyzer from a Lex file:
+- Generate an analyzer from a Lex file, storing it to the file `lex.yy.c`:
 
 `lex {{analyzer.l}}`
 
+- Write analyzer to `s`[`t`]`dout`:
```
Comment (kbdharun) [inline pages/linux/lex.md:11]:
```suggestion
- Write analyzer to `stdout`:
```

**Review by gutjuri — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Comment by kbdharun** [comment]
@sebastiaanspeck The updated check PR script still seems to have some issues.

**Comment by kbdharun** [comment]
> What exactly is the issue you try to point at?

Currently, there are two output prompts:

```
The page pages/common/lex.md already exists under the linux platform.
The page pages/linux/lex.md already exists under the common platform.
```

IMO we should only tell linux/other pages already exists under common and not the other way around. But IG it's fine if others don't have an issue with the current behavior

**Review by kbdharun — APPROVED** [review]

**Pull request merged**